elyra/pipeline/component_registry.py

```diff
@@ -102,7 +103,7 @@ def get_generic_component(component_id: str) -> Component:
         return ComponentRegistry._generic_components.get(component_id)
 
     @staticmethod
-    def to_canvas_palette(components: List[Component]) -> dict:
+    def to_canvas_palette(components: List[Component]) -> str:
```

Comment:
Just a heads up that #1882 is returning a dict (and calling `json.loads()`) in the method.  Good catch on the invalid hint though.  Same goes for `to_canvas_properties()` below.

Reply:
got it, rebased on 1882 and changing it back